packages/medusa/src/api/routes/admin/price-lists/delete-variant-prices.ts

```diff
@@ -59,16 +61,44 @@ export default async (req, res) => {
 
   const priceListService: PriceListService =
     req.scope.resolve("priceListService")
-
+  const featureFlagRouter: FlagRouter = req.scope.resolve("featureFlagRouter")
   const manager: EntityManager = req.scope.resolve("manager")
-  const [deletedPriceIds] = await manager.transaction(
-    async (transactionManager) => {
-      return await priceListService
-        .withTransaction(transactionManager)
-        .deleteVariantPrices(id, [variant_id])
-    }
+
+  const isMedusaV2FlagEnabled = featureFlagRouter.isFeatureEnabled(
+    MedusaV2Flag.key
   )
 
+  let deletedPriceIds: string[] = []
+
+  if (isMedusaV2FlagEnabled) {
+    const deletePriceListProductsWorkflow = removePriceListVariantPrices(
+      req.scope
+    )
+
+    const input = {
+      variantIds: [variant_id],
+      priceListId: id,
+    }
+
+    const { result } = await deletePriceListProductsWorkflow.run({
+      input,
+      context: {
+        manager,
+      },
+    })
+    deletedPriceIds = result
+  } else {
+    const [deletedIds] = await manager.transaction(
+      async (transactionManager) => {
+        return await priceListService
+          .withTransaction(transactionManager)
+          .deleteVariantPrices(id, [variant_id])
+      }
+    )
+
+    deletedPriceIds = deletedIds
```

Comment:
comment: same comments as the previous files are valid

Reply:
same as above